esmvalcore/preprocessor/_regrid_unstructured.py

```diff
@@ -47,3 +59,309 @@ def regridder(
+                f"on the same source grid as this regridder"
+            )
+
+        # Get regridded data
+        regridded_data = self._get_regridded_data(cube)
+
+        # Get coordinates of regridded cube
+
+        # (1) New dimensional coordinates are the ones from the source cube
+        # (excluding the unstructured grid dimension) plus the (x, y) target
+        # grid dimensions. All dimensions to the right of the unstructured grid
+        # dimension need to be shifted to the right by 1.
+        udim = cube.coord_dims('latitude')[0]
+        dim_coords_and_dims = [
+            (c, cube.coord_dims(c)[0]) for c in cube.coords(dim_coords=True) if
+            udim not in cube.coord_dims(c)
+        ]
+        dim_coords_and_dims = [
+            (c, d) if d < udim else (c, d + 1) for (c, d) in
+            dim_coords_and_dims
+        ]
+        dim_coords_and_dims.append((self.tgt_coords[0], udim))
+        dim_coords_and_dims.append((self.tgt_coords[1], udim + 1))
+
+        # (2) Include all auxiliary coordinates that do not span unstructured
+        # grid dimension (also make sure to shift all dimensions to the right
+        # of the unstructured grid to the right by 1)
+        old_aux_coords_and_dims = [
+            (c, cube.coord_dims(c)) for c in cube.coords(dim_coords=False) if
+            udim not in cube.coord_dims(c)
+        ]
+        aux_coords_and_dims = []
+        for (aux_coord, dims) in old_aux_coords_and_dims:
+            dims = tuple(d if d < udim else d + 1 for d in dims)
+            aux_coords_and_dims.append((aux_coord, dims))
+
+        # Create new cube
+        regridded_cube = Cube(
+            regridded_data,
+            dim_coords_and_dims=dim_coords_and_dims,
+            aux_coords_and_dims=aux_coords_and_dims,
+        )
+        regridded_cube.metadata = cube.metadata
+
+        return regridded_cube
+
+    def _get_regridded_data(self, cube: Cube) -> np.ndarray | da.Array:
+        """Get regridded data."""
+        udim = cube.coord_dims('latitude')[0]
+
+        # Cube must not be chunked along latitude and longitude dimension
+        rechunk_cube(cube, ['latitude', 'longitude'])
+
+        # Make sure that masked arrays are filled with nan's so they are
+        # handled properly
+        npx = da if cube.has_lazy_data() else np
+        src_data = npx.ma.filled(cube.core_data(), np.nan)
+
+        shape = [s for (d, s) in enumerate(cube.shape) if d != udim]
+        shape.insert(udim, self.tgt_coords[0].core_points().size)
+        shape.insert(udim + 1, self.tgt_coords[1].core_points().size)
+        new_shape = tuple(shape)
+
+        # Perform actual regridding and assign correct mask
+        regridded_data: np.ndarray | da.Array
+        if cube.has_lazy_data():
+            regridded_data = self._regrid_lazy(
+                src_data, udim, self.weights.dtype
+            )
+            # TODO: add limit='128 MiB' to reshape once dask bug is solved
```

Comment:
I would recommend respecting the configured chunk-size instead of hardcoding 128 MiB

Reply:
Good point. The [doc of reshape](https://docs.dask.org/en/stable/generated/dask.array.reshape.html) suggests that this is done for `limit=None`. However, this may raise the following warning:

```bash
PerformanceWarning: Reshaping is producing a large chunk. To accept the large
chunk and silence this warning, set the option
    >>> with dask.config.set(**{'array.slicing.split_large_chunks': False}):
    ...     array.reshape(shape)

To avoid creating the large chunks, set the option
    >>> with dask.config.set(**{'array.slicing.split_large_chunks': True}):
    ...     array.reshape(shape)Explictly passing ``limit`` to ``reshape`` will also silence this warning
    >>> array.reshape(shape, limit='128 MiB')
  return function(items, **kwargs)
```

Reply:
Would it be possible to do the reshape as part of the regridding instead of afterwards, to avoid this issue? Maybe in the `_interpolate` function?

Reply:
Great idea! Implemented that in https://github.com/ESMValGroup/ESMValCore/pull/2350/commits/fa52c672062c8cfb74b3da9d5d227ef76632bb9c